specreduce/background.py

```diff
@@ -0,0 +1,205 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from dataclasses import dataclass, field
+
+import numpy as np
+from astropy.nddata import CCDData
+
+from specreduce.extract import _ap_weight_image
+from specreduce.tracing import Trace, FlatTrace
+
+__all__ = ['Background']
+
+
+@dataclass
+class Background:
+    """
+    Determine the background from an image for subtraction
+
+    Parameters
+    ----------
+    image : nddata-compatible image
+        image with 2-D spectral image data
+    traces : List
+        list of trace objects (or integers to define FlatTraces) to
+        extract the background
+    width : float
+        width of extraction aperture in pixels
+    statistic: string
+        statistic to use when computing the background.  'average' will
+        account for partial pixel weights, 'median' will include all partial
+        pixels.
+    disp_axis : int
+        dispersion axis
+    crossdisp_axis : int
+        cross-dispersion axis
+    """
+    # required so numpy won't call __rsub__ on individual elements
+    # https://stackoverflow.com/a/58409215
+    __array_ufunc__ = None
+
+    image: CCDData
+    traces: list = field(default_factory=list)
+    width: float = 5
+    statistic: str = 'average'
+    disp_axis: int = 1
+    crossdisp_axis: int = 0
```

Comment:
only `disp_axis` is needed as an input. `crossdisp_axis` can be defined based on that within `__post_init__`.

Reply:
I agree, but see that the other recent classes also define `crossdisp_axis` and some of this logic may move up to the trace objects in the future. Do you have a preference for whether to do that just in this PR or across all classes at once (or maybe inheriting from a base class/mixin that handles that)?

Reply:
technically, if you're accepting both as inputs, there should be validation to make sure they're not equal and that both are in range of the images dimensions. ultimately, `disp_axis` should be an attribute of `Trace` classes and flow down from there as described in #89. so i guess it's fine to leave as-is for now and do all the refactoring then.

Reply:
I like that plan.  I agree that it should be changed, but that #89 would be the best opportunity to handle it consistently across all recent work.

Reply:
i have a proto-PR in the works to cover that and more, but won't be ready in time for next week's release.